src/context/GlobalProvider.tsx

```diff
@@ -1,25 +1,97 @@
-import React, { createContext, useState } from 'react';
+import React, { createContext } from 'react';
+import { CartItem } from '../types/CartItem';
+import { FavoriteItem } from '../types/FavoriteItem';
+import { useCartItems } from './hooks/useCartItems';
+import { useFavoriteItems } from './hooks/useFavoriteItems';
 
-enum Theme {
-  Light = 'light',
-  Dark = 'dark',
+enum QuantityOperation {
+  Increase = 'increase',
+  Decrease = 'decrease',
 }
 
-interface PhonesContextProps {
-  theme: Theme;
-  setTheme: React.Dispatch<React.SetStateAction<Theme>>;
-}
+type DefaultValuesType = {
+  cartItems: CartItem[] | FavoriteItem[];
+  addToCart: (id: number) => void;
+  removeFromCart: (id: number) => void;
+  changeQuantity: (id: number, operation: QuantityOperation) => void;
+  favoriteItems: CartItem[] | FavoriteItem[];
+  handleFavorite: (id: number) => void;
+};
 
-const PhonesContext = createContext<PhonesContextProps | undefined>(undefined);
+export const PhonesContext = createContext<DefaultValuesType>({
+  cartItems: [],
+  addToCart: () => {},
+  removeFromCart: () => {},
+  changeQuantity: () => {},
+  favoriteItems: [],
+  handleFavorite: () => {},
+});
 
 export const PhonesProvider: React.FC<{ children: React.ReactNode }> = ({
   children,
 }) => {
-  const [theme, setTheme] = useState(Theme.Light);
+  const [cartItems, setCartItems] = useCartItems([]);
+  const [favoriteItems, setFavoriteItems] = useFavoriteItems([]);
+
+  const addToCart = (id: number) => {
+    const newRecord: CartItem = {
+      id,
+      quantity: 1,
+    };
+
+    setCartItems([...cartItems, newRecord]);
+  };
+
+  const removeFromCart = (id: number) => {
+    const updatedCartItems = cartItems.filter((item) => item.id !== id);
+
+    setCartItems(updatedCartItems);
+  };
+
+  const changeQuantity = (id: number, operation: QuantityOperation) => {
+    const updatedCartItems = cartItems.map((item) => {
+      if (item.id === id) {
+        switch (operation) {
+          case QuantityOperation.Increase:
+            return {
+              ...item,
+              quantity: item.quantity + 1,
+            };
+          case QuantityOperation.Decrease:
+            return {
+              ...item,
+              quantity: item.quantity - 1,
+            };
+          default:
+            return item;
+        }
+      }
+
+      return item;
+    });
+
+    setCartItems(updatedCartItems);
+  };
+
+  const handleFavorite = (id: number) => {
+    const udpatedItems = [...favoriteItems];
```

Comment:
I'm not sure because of the repetition, but maybe you meant to call it updatedItems?